### PR TITLE
Fix client component imports are not being correctly tracked

### LINKED ISSUE #7: Non-page server component cannot import client component
It seems that client components can be only directly imported from a page component. 

For example, the `News` component located in `rsc.server.js` can import the `Story` client component just fine. If we, however, move the `StoryWithData` to a separate file (with the `.server` suffix or none i.e. a shared component), Next.js returns errors.

Here's a diff off a branch from my fork that shows this issue: 
https://github.com/vercel/next-rsc-demo/compare/main...zaiste:non-page-server-comp-cannot-import-client-comp

### PR DESCRIPTION
**Pull request by shuding**
When there're multi-level server component imports, some components are eliminated after going through the client compilation, because we are only tracking `.client.` dependencies inside `.server.` components. Instead we should keep all imports for the client compilation, but only remove the content of server components.

This fixes https://github.com/vercel/next-rsc-demo/issues/7.

## Bug

- [x] Related issues linked using `fixes #number`
- [x] Integration tests added
- [ ] Errors have helpful link attached, see `contributing.md`

## Feature

- [ ] Implements an existing feature request or RFC. Make sure the feature request has been accepted for implementation before opening a PR.
- [ ] Related issues linked using `fixes #number`
- [ ] Integration tests added
- [ ] Documentation added
- [ ] Telemetry added. In case of a feature if it's used or not.
- [ ] Errors have helpful link attached, see `contributing.md`

## Documentation / Examples

- [ ] Make sure the linting passes by running `yarn lint`

### CODE REVIEW

File: packages/next/build/webpack-config.ts [hunk 1]
```diff
@@ -1187,7 +1187,7 @@ export default async function getBaseWebpackConfig(
               },
             ]
           : []),
-        ...(webServerRuntime && hasServerComponents
+        ...(hasServerComponents && webServerRuntime
```
Comment (shuding) [inline packages/next/build/webpack-config.ts:1190]: Swapped the conditions for better consistency with the other loader.

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Build</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | shuding/next.js shu/04e1 | Change |
| - | - | - | - |
| buildDuration | 18s | 17.7s | -313ms |
| buildDurationCached | 3.9s | 3.8s | -47ms |
| nodeModulesSize | 293 MB | 293 MB | ⚠️ +213 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | shuding/next.js shu/04e1 | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.855 | 2.781 | -0.07 |
| / avg req/sec | 875.75 | 899.04 | +23.29 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.305 | 1.304 | 0 |
| /error-in-render avg req/sec | 1916.03 | 1917.84 | +1.81 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | shuding/next.js shu/04e1 | Change |
| - | - | - | - |
| 450.HASH.js gzip | 179 B | 179 B | ✓ |
| framework-HASH.js gzip | 42.2 kB | 42.2 kB | ✓ |
| main-HASH.js gzip | 28 kB | 28 kB | ✓ |
| webpack-HASH.js gzip | 1.45 kB | 1.45 kB | ✓ |
| Overall change | 71.9 kB | 71.9 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | shuding/next.js shu/04e1 | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | shuding/next.js shu/04e1 | Change |
| - | - | - | - |
| _app-HASH.js gzip | 1.23 kB | 1.23 kB | ✓ |
| _error-HASH.js gzip | 194 B | 194 B | ✓ |
| amp-HASH.js gzip | 312 B | 312 B | ✓ |
| css-HASH.js gzip | 327 B | 327 B | ✓ |
| dynamic-HASH.js gzip | 2.38 kB | 2.38 kB | ✓ |
| head-HASH.js gzip | 350 B | 350 B | ✓ |
| hooks-HASH.js gzip | 635 B | 635 B | ✓ |
| image-HASH.js gzip | 4.44 kB | 4.44 kB | ✓ |
| index-HASH.js gzip | 263 B | 263 B | ✓ |
| link-HASH.js gzip | 1.87 kB | 1.87 kB | ✓ |
| routerDirect..HASH.js gzip | 321 B | 321 B | ✓ |
| script-HASH.js gzip | 383 B | 383 B | ✓ |
| withRouter-HASH.js gzip | 318 B | 318 B | ✓ |
| 334f979574ae..6f4.css gzip | 106 B | 106 B | ✓ |
| Overall change | 13.1 kB | 13.1 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | shuding/next.js shu/04e1 | Change |
| - | - | - | - |
| _buildManifest.js gzip | 459 B | 459 B | ✓ |
| Overall change | 459 B | 459 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | shuding/next.js shu/04e1 | Change |
| - | - | - | - |
| index.html gzip | 521 B | 521 B | ✓ |
| link.html gzip | 533 B | 533 B | ✓ |
| withRouter.html gzip | 515 B | 515 B | ✓ |
| Overall change | 1.57 kB | 1.57 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Default Build with SWC</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | shuding/next.js shu/04e1 | Change |
| - | - | - | - |
| buildDuration | 19.3s | 19.4s | ⚠️ +57ms |
| buildDurationCached | 3.8s | 4s | ⚠️ +144ms |
| nodeModulesSize | 293 MB | 293 MB | ⚠️ +213 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall decrease ⚠️</summary>

|  | vercel/next.js canary  | shuding/next.js shu/04e1 | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.797 | 2.839 | ⚠️ +0.04 |
| / avg req/sec | 893.71 | 880.56 | ⚠️ -13.15 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.324 | 1.383 | ⚠️ +0.06 |
| /error-in-render avg req/sec | 1887.65 | 1808.13 | ⚠️ -79.52 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | shuding/next.js shu/04e1 | Change |
| - | - | - | - |
| 450.HASH.js gzip | 179 B | 179 B | ✓ |
| framework-HASH.js gzip | 42.3 kB | 42.3 kB | ✓ |
| main-HASH.js gzip | 28.2 kB | 28.2 kB | ✓ |
| webpack-HASH.js gzip | 1.43 kB | 1.43 kB | ✓ |
| Overall change | 72.1 kB | 72.1 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | shuding/next.js shu/04e1 | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | shuding/next.js shu/04e1 | Change |
| - | - | - | - |
| _app-HASH.js gzip | 1.22 kB | 1.22 kB | ✓ |
| _error-HASH.js gzip | 180 B | 180 B | ✓ |
| amp-HASH.js gzip | 305 B | 305 B | ✓ |
| css-HASH.js gzip | 321 B | 321 B | ✓ |
| dynamic-HASH.js gzip | 2.38 kB | 2.38 kB | ✓ |
| head-HASH.js gzip | 342 B | 342 B | ✓ |
| hooks-HASH.js gzip | 622 B | 622 B | ✓ |
| image-HASH.js gzip | 4.46 kB | 4.46 kB | ✓ |
| index-HASH.js gzip | 256 B | 256 B | ✓ |
| link-HASH.js gzip | 1.91 kB | 1.91 kB | ✓ |
| routerDirect..HASH.js gzip | 314 B | 314 B | ✓ |
| script-HASH.js gzip | 375 B | 375 B | ✓ |
| withRouter-HASH.js gzip | 309 B | 309 B | ✓ |
| 334f979574ae..6f4.css gzip | 106 B | 106 B | ✓ |
| Overall change | 13.1 kB | 13.1 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | shuding/next.js shu/04e1 | Change |
| - | - | - | - |
| _buildManifest.js gzip | 460 B | 460 B | ✓ |
| Overall change | 460 B | 460 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | shuding/next.js shu/04e1 | Change |
| - | - | - | - |
| index.html gzip | 521 B | 521 B | ✓ |
| link.html gzip | 534 B | 534 B | ✓ |
| withRouter.html gzip | 516 B | 516 B | ✓ |
| Overall change | 1.57 kB | 1.57 kB | ✓ |

</details>

</details>
Commit: 62b052d5a74e1a3573fdf923fc2647bce645b38d

**Comment by ijjk** [comment]
## Failing test suites 
Commit: 62b052d5a74e1a3573fdf923fc2647bce645b38d

**test/integration/react-streaming-and-server-components/test/index.test.js** 
- concurrentFeatures - dev > should support multi-level server component imports
- concurrentFeatures - prod > should support multi-level server component imports

<details>
<summary>Expand output</summary>

  ● concurrentFeatures - prod › should support multi-level server component imports

    expect(received).toContain(expected) // indexOf

    Expected substring: "bar.server.js: foo.client"
    Received string:    "<!DOCTYPE html><html><head><meta name=\"next-font-preconnect\"/><meta charSet=\"utf-8\"/><meta name=\"viewport\" content=\"width=device-width\"/><meta name=\"next-head-count\" content=\"2\"/><noscript data-n-css=\"\"></noscript><script defer=\"\" nomodule=\"\" src=\"/_next/static/chunks/polyfills-5cd94c89d3acac5f.js\"></script><script src=\"/_next/static/chunks/webpack-785eee2fec18f6ec.js\" defer=\"\"></script><script src=\"/_next/static/chunks/framework-c6501f8788ee8509.js\" defer=\"\"></script><script src=\"/_next/static/chunks/main-93f4865b267d4d1d.js\" defer=\"\"></script><script src=\"/_next/static/chunks/pages/_app-47a86cae646d4088.js\" defer=\"\"></script><script src=\"/_next/static/chunks/pages/multi-91a334093ecc03bd.js\" defer=\"\"></script><script src=\"/_next/static/Co-DxtEBjWaX6Hvbzn_gC/_buildManifest.js\" defer=\"\"></script><script src=\"/_next/static/Co-DxtEBjWaX6Hvbzn_gC/_ssgManifest.js\" defer=\"\"></script><script src=\"/_next/static/Co-DxtEBjWaX6Hvbzn_gC/_middlewareManifest.js\" defer=\"\"></script></head><body><div id=\"__next\"><template id=\"B:0\"></template><div hidden id=\"S:0\"><template id=\"P:1\"></template></div><div hidden id=\"S:1\"><div>bar.server.js: foo.client</div></div><script>function $RS(a,b){a=document.getElementById(a);b=document.getElementById(b);for(a.parentNode.removeChild(a);a.firstChild;)b.parentNode.insertBefore(a.firstChild,b);b.parentNode.removeChild(b)};$RS(\"S:1\",\"P:1\")</script><script>function $RC(a,b){a=document.getElementById(a);b=document.getElementById(b);b.parentNode.removeChild(b);if(a){a=a.previousSibling;var f=a.parentNode,c=a.nextSibling,e=0;do{if(c&&8===c.nodeType){var d=c.data;if(\"/$\"===d)if(0===e)break;else e--;else\"$\"!==d&&\"$?\"!==d&&\"$!\"!==d||e++}d=c.nextSibling;f.removeChild(c);c=d}while(c);for(;b.firstChild;)f.insertBefore(b.firstChild,c);a.data=\"$\";a._reactRetry&&a._reactRetry()}};$RC(\"B:0\",\"S:0\")</script></div><script id=\"__NEXT_DATA__\" type=\"application/json\">{\"props\":{\"pageProps\":{}},\"page\":\"/multi\",\"query\":{},\"buildId\":\"Co-DxtEBjWaX6Hvbzn_gC\",\"autoExport\":true,\"isFallback\":false,\"rsc\":true,\"scriptLoader\":[]}</script></body></html>"

      250 |     })
      251 |
    > 252 |     runTests(context)
          |              ^
      253 |   })
      254 | }
      255 |

      at Object.<anonymous> (integration/react-streaming-and-server-components/test/index.test.js:252:14)

  ● concurrentFeatures - dev › should support multi-level server component imports

    expect(received).toContain(expected) // indexOf

    Expected substring: "bar.server.js: foo.client"
    Received string:    "<!DOCTYPE html><html><head><style data-next-hide-fouc=\"true\">body{display:none}</style><noscript data-next-hide-fouc=\"true\"><style>body{display:block}</style></noscript><meta charSet=\"utf-8\"/><meta name=\"viewport\" content=\"width=device-width\"/><meta name=\"next-head-count\" content=\"2\"/><noscript data-n-css=\"\"></noscript><script defer=\"\" nomodule=\"\" src=\"/_next/static/chunks/polyfills.js?ts=1635895394326\"></script><script src=\"/_next/static/chunks/webpack.js?ts=1635895394326\" defer=\"\"></script><script src=\"/_next/static/chunks/main.js?ts=1635895394326\" defer=\"\"></script><script src=\"/_next/static/chunks/pages/_app.js?ts=1635895394326\" defer=\"\"></script><script src=\"/_next/static/chunks/pages/multi.js?ts=1635895394326\" defer=\"\"></script><script src=\"/_next/static/development/_buildManifest.js?ts=1635895394326\" defer=\"\"></script><script src=\"/_next/static/development/_ssgManifest.js?ts=1635895394326\" defer=\"\"></script><script src=\"/_next/static/development/_middlewareManifest.js?ts=1635895394326\" defer=\"\"></script><noscript id=\"__next_css__DO_NOT_USE__\"></noscript></head><body><div id=\"__next\"><template id=\"B:0\"></template><div hidden id=\"S:0\"><template id=\"P:1\"></template></div><div hidden id=\"S:1\"><div>bar.server.js: foo.client</div></div><script>function $RS(a,b){a=document.getElementById(a);b=document.getElementById(b);for(a.parentNode.removeChild(a);a.firstChild;)b.parentNode.insertBefore(a.firstChild,b);b.parentNode.removeChild(b)};$RS(\"S:1\",\"P:1\")</script><script>function $RC(a,b){a=document.getElementById(a);b=document.getElementById(b);b.parentNode.removeChild(b);if(a){a=a.previousSibling;var f=a.parentNode,c=a.nextSibling,e=0;do{if(c&&8===c.nodeType){var d=c.data;if(\"/$\"===d)if(0===e)break;else e--;else\"$\"!==d&&\"$?\"!==d&&\"$!\"!==d||e++}d=c.nextSibling;f.removeChild(c);c=d}while(c);for(;b.firstChild;)f.insertBefore(b.firstChild,c);a.data=\"$\";a._reactRetry&&a._reactRetry()}};$RC(\"B:0\",\"S:0\")</script></div><script src=\"/_next/static/chunks/react-refresh.js?ts=1635895394326\"></script><script id=\"__NEXT_DATA__\" type=\"application/json\">{\"props\":{\"pageProps\":{}},\"page\":\"/multi\",\"query\":{},\"buildId\":\"development\",\"nextExport\":true,\"autoExport\":true,\"isFallback\":false,\"rsc\":true,\"scriptLoader\":[]}</script></body></html>"

      250 |     })
      251 |
    > 252 |     runTests(context)
          |              ^
      253 |   })
      254 | }
      255 |

      at Object.<anonymous> (integration/react-streaming-and-server-components/test/index.test.js:252:14)

</details>

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Build</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | shuding/next.js shu/04e1 | Change |
| - | - | - | - |
| buildDuration | 23.9s | 23.9s | ⚠️ +53ms |
| buildDurationCached | 5.1s | 5s | -83ms |
| nodeModulesSize | 293 MB | 293 MB | ⚠️ +220 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall decrease ⚠️</summary>

|  | vercel/next.js canary  | shuding/next.js shu/04e1 | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 3.951 | 3.914 | -0.04 |
| / avg req/sec | 632.67 | 638.66 | +5.99 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 2.037 | 2.113 | ⚠️ +0.08 |
| /error-in-render avg req/sec | 1227.53 | 1183.27 | ⚠️ -44.26 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | shuding/next.js shu/04e1 | Change |
| - | - | - | - |
| 450.HASH.js gzip | 179 B | 179 B | ✓ |
| framework-HASH.js gzip | 42.2 kB | 42.2 kB | ✓ |
| main-HASH.js gzip | 28 kB | 28 kB | ✓ |
| webpack-HASH.js gzip | 1.45 kB | 1.45 kB | ✓ |
| Overall change | 71.9 kB | 71.9 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | shuding/next.js shu/04e1 | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | shuding/next.js shu/04e1 | Change |
| - | - | - | - |
| _app-HASH.js gzip | 1.23 kB | 1.23 kB | ✓ |
| _error-HASH.js gzip | 194 B | 194 B | ✓ |
| amp-HASH.js gzip | 312 B | 312 B | ✓ |
| css-HASH.js gzip | 327 B | 327 B | ✓ |
| dynamic-HASH.js gzip | 2.38 kB | 2.38 kB | ✓ |
| head-HASH.js gzip | 350 B | 350 B | ✓ |
| hooks-HASH.js gzip | 635 B | 635 B | ✓ |
| image-HASH.js gzip | 4.44 kB | 4.44 kB | ✓ |
| index-HASH.js gzip | 263 B | 263 B | ✓ |
| link-HASH.js gzip | 1.87 kB | 1.87 kB | ✓ |
| routerDirect..HASH.js gzip | 321 B | 321 B | ✓ |
| script-HASH.js gzip | 383 B | 383 B | ✓ |
| withRouter-HASH.js gzip | 318 B | 318 B | ✓ |
| 334f979574ae..6f4.css gzip | 106 B | 106 B | ✓ |
| Overall change | 13.1 kB | 13.1 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | shuding/next.js shu/04e1 | Change |
| - | - | - | - |
| _buildManifest.js gzip | 459 B | 459 B | ✓ |
| Overall change | 459 B | 459 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | shuding/next.js shu/04e1 | Change |
| - | - | - | - |
| index.html gzip | 521 B | 521 B | ✓ |
| link.html gzip | 533 B | 533 B | ✓ |
| withRouter.html gzip | 515 B | 515 B | ✓ |
| Overall change | 1.57 kB | 1.57 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Default Build with SWC</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | shuding/next.js shu/04e1 | Change |
| - | - | - | - |
| buildDuration | 25.9s | 25.7s | -248ms |
| buildDurationCached | 5s | 5s | -33ms |
| nodeModulesSize | 293 MB | 293 MB | ⚠️ +220 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall decrease ⚠️</summary>

|  | vercel/next.js canary  | shuding/next.js shu/04e1 | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 4.068 | 4.069 | 0 |
| / avg req/sec | 614.58 | 614.45 | ⚠️ -0.13 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 2.058 | 2.09 | ⚠️ +0.03 |
| /error-in-render avg req/sec | 1214.66 | 1196.37 | ⚠️ -18.29 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | shuding/next.js shu/04e1 | Change |
| - | - | - | - |
| 450.HASH.js gzip | 179 B | 179 B | ✓ |
| framework-HASH.js gzip | 42.3 kB | 42.3 kB | ✓ |
| main-HASH.js gzip | 28.2 kB | 28.2 kB | ✓ |
| webpack-HASH.js gzip | 1.43 kB | 1.43 kB | ✓ |
| Overall change | 72.1 kB | 72.1 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | shuding/next.js shu/04e1 | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | shuding/next.js shu/04e1 | Change |
| - | - | - | - |
| _app-HASH.js gzip | 1.22 kB | 1.22 kB | ✓ |
| _error-HASH.js gzip | 180 B | 180 B | ✓ |
| amp-HASH.js gzip | 305 B | 305 B | ✓ |
| css-HASH.js gzip | 321 B | 321 B | ✓ |
| dynamic-HASH.js gzip | 2.38 kB | 2.38 kB | ✓ |
| head-HASH.js gzip | 342 B | 342 B | ✓ |
| hooks-HASH.js gzip | 622 B | 622 B | ✓ |
| image-HASH.js gzip | 4.46 kB | 4.46 kB | ✓ |
| index-HASH.js gzip | 256 B | 256 B | ✓ |
| link-HASH.js gzip | 1.91 kB | 1.91 kB | ✓ |
| routerDirect..HASH.js gzip | 314 B | 314 B | ✓ |
| script-HASH.js gzip | 375 B | 375 B | ✓ |
| withRouter-HASH.js gzip | 309 B | 309 B | ✓ |
| 334f979574ae..6f4.css gzip | 106 B | 106 B | ✓ |
| Overall change | 13.1 kB | 13.1 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | shuding/next.js shu/04e1 | Change |
| - | - | - | - |
| _buildManifest.js gzip | 460 B | 460 B | ✓ |
| Overall change | 460 B | 460 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | shuding/next.js shu/04e1 | Change |
| - | - | - | - |
| index.html gzip | 521 B | 521 B | ✓ |
| link.html gzip | 534 B | 534 B | ✓ |
| withRouter.html gzip | 516 B | 516 B | ✓ |
| Overall change | 1.57 kB | 1.57 kB | ✓ |

</details>

</details>
Commit: ad8de4b8c92133ffc4f4602119f90557e9a197fe

**Review by ijjk — APPROVED** [review]

**Comment by ijjk** [comment]
## Failing test suites 
Commit: 19f017699d4bba0c20dd94c845971e18e49701f3

**test/development/basic/hmr.test.ts** 
- basic HMR > Hot Module Reloading > editing a page > should update styles correctly

<details>
<summary>Expand output</summary>

  ● basic HMR › Hot Module Reloading › editing a page › should update styles correctly

    page.waitForSelector: Timeout 30000ms exceeded.
    =========================== logs ===========================
    waiting for selector ".hmr-style-page p"
    ============================================================

      250 |     return this.chain(() => {
      251 |       return page
    > 252 |         .waitForSelector(selector, { timeout, state: 'attached' })
          |          ^
      253 |         .then(async (el) => {
      254 |           // it seems selenium waits longer and tests rely on this behavior
      255 |           // so we wait for the load event fire before returning

      at lib/browsers/playwright.ts:252:10

</details>

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Build</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | shuding/next.js shu/04e1 | Change |
| - | - | - | - |
| buildDuration | 19.8s | 20.2s | ⚠️ +323ms |
| buildDurationCached | 4.3s | 4.3s | ⚠️ +76ms |
| nodeModulesSize | 293 MB | 293 MB | ⚠️ +227 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | shuding/next.js shu/04e1 | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 3.514 | 3.768 | ⚠️ +0.25 |
| / avg req/sec | 711.49 | 663.47 | ⚠️ -48.02 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 2.047 | 1.849 | -0.2 |
| /error-in-render avg req/sec | 1221.46 | 1352.26 | +130.8 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | shuding/next.js shu/04e1 | Change |
| - | - | - | - |
| 450.HASH.js gzip | 179 B | 179 B | ✓ |
| framework-HASH.js gzip | 42.2 kB | 42.2 kB | ✓ |
| main-HASH.js gzip | 28 kB | 28 kB | ✓ |
| webpack-HASH.js gzip | 1.45 kB | 1.45 kB | ✓ |
| Overall change | 71.9 kB | 71.9 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | shuding/next.js shu/04e1 | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | shuding/next.js shu/04e1 | Change |
| - | - | - | - |
| _app-HASH.js gzip | 1.23 kB | 1.23 kB | ✓ |
| _error-HASH.js gzip | 194 B | 194 B | ✓ |
| amp-HASH.js gzip | 312 B | 312 B | ✓ |
| css-HASH.js gzip | 327 B | 327 B | ✓ |
| dynamic-HASH.js gzip | 2.38 kB | 2.38 kB | ✓ |
| head-HASH.js gzip | 350 B | 350 B | ✓ |
| hooks-HASH.js gzip | 635 B | 635 B | ✓ |
| image-HASH.js gzip | 4.44 kB | 4.44 kB | ✓ |
| index-HASH.js gzip | 263 B | 263 B | ✓ |
| link-HASH.js gzip | 1.87 kB | 1.87 kB | ✓ |
| routerDirect..HASH.js gzip | 321 B | 321 B | ✓ |
| script-HASH.js gzip | 383 B | 383 B | ✓ |
| withRouter-HASH.js gzip | 318 B | 318 B | ✓ |
| 334f979574ae..6f4.css gzip | 106 B | 106 B | ✓ |
| Overall change | 13.1 kB | 13.1 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | shuding/next.js shu/04e1 | Change |
| - | - | - | - |
| _buildManifest.js gzip | 459 B | 459 B | ✓ |
| Overall change | 459 B | 459 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | shuding/next.js shu/04e1 | Change |
| - | - | - | - |
| index.html gzip | 521 B | 521 B | ✓ |
| link.html gzip | 533 B | 533 B | ✓ |
| withRouter.html gzip | 515 B | 515 B | ✓ |
| Overall change | 1.57 kB | 1.57 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Default Build with SWC</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | shuding/next.js shu/04e1 | Change |
| - | - | - | - |
| buildDuration | 21.8s | 21.6s | -216ms |
| buildDurationCached | 4.2s | 4.3s | ⚠️ +56ms |
| nodeModulesSize | 293 MB | 293 MB | ⚠️ +227 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | shuding/next.js shu/04e1 | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 3.343 | 3.495 | ⚠️ +0.15 |
| / avg req/sec | 747.8 | 715.27 | ⚠️ -32.53 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.903 | 1.835 | -0.07 |
| /error-in-render avg req/sec | 1313.95 | 1362.76 | +48.81 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | shuding/next.js shu/04e1 | Change |
| - | - | - | - |
| 450.HASH.js gzip | 179 B | 179 B | ✓ |
| framework-HASH.js gzip | 42.3 kB | 42.3 kB | ✓ |
| main-HASH.js gzip | 28.2 kB | 28.2 kB | ✓ |
| webpack-HASH.js gzip | 1.43 kB | 1.43 kB | ✓ |
| Overall change | 72.1 kB | 72.1 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | shuding/next.js shu/04e1 | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | shuding/next.js shu/04e1 | Change |
| - | - | - | - |
| _app-HASH.js gzip | 1.22 kB | 1.22 kB | ✓ |
| _error-HASH.js gzip | 180 B | 180 B | ✓ |
| amp-HASH.js gzip | 305 B | 305 B | ✓ |
| css-HASH.js gzip | 321 B | 321 B | ✓ |
| dynamic-HASH.js gzip | 2.38 kB | 2.38 kB | ✓ |
| head-HASH.js gzip | 342 B | 342 B | ✓ |
| hooks-HASH.js gzip | 622 B | 622 B | ✓ |
| image-HASH.js gzip | 4.46 kB | 4.46 kB | ✓ |
| index-HASH.js gzip | 256 B | 256 B | ✓ |
| link-HASH.js gzip | 1.91 kB | 1.91 kB | ✓ |
| routerDirect..HASH.js gzip | 314 B | 314 B | ✓ |
| script-HASH.js gzip | 375 B | 375 B | ✓ |
| withRouter-HASH.js gzip | 309 B | 309 B | ✓ |
| 334f979574ae..6f4.css gzip | 106 B | 106 B | ✓ |
| Overall change | 13.1 kB | 13.1 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | shuding/next.js shu/04e1 | Change |
| - | - | - | - |
| _buildManifest.js gzip | 460 B | 460 B | ✓ |
| Overall change | 460 B | 460 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | shuding/next.js shu/04e1 | Change |
| - | - | - | - |
| index.html gzip | 521 B | 521 B | ✓ |
| link.html gzip | 534 B | 534 B | ✓ |
| withRouter.html gzip | 516 B | 516 B | ✓ |
| Overall change | 1.57 kB | 1.57 kB | ✓ |

</details>

</details>
Commit: 19f017699d4bba0c20dd94c845971e18e49701f3

**Pull request merged**